### ISSUE TITLE
CMD Scripts doesn't execute

### ISSUE THREAD

**Issue by geghamyan** [appveyor/ci#3424]
**Environment:**
* BYOC systems where the "PATH" variables is modified. A path to an object with 'cmd' name is preceding the %SystemRoot%.
Observed Behavior:
* AppVeyor build agent doesn't execute 'cmd' scripts.

**Example:**
Because of the installation of MSYS or other components or edited manually, the %PATH% variable looks like this
`PATH=...;C:\MinGW\msys\1.0\bin;C:\MinGW\bin;...;%SystemRoot%\system32;%SystemRoot...`

In command line the where.exe command gives the following result:
```
C:\Users\user>where cmd
C:\MinGW\msys\1.0\bin\cmd
C:\Windows\System32\cmd.exe
```

It looks like the build agent internally tries to use "where cmd" command to locate the "cmd.exe" command interpreter and takes the first result which is incorrect.
I observe the following in procmon on the BYOC machine:
```
2:14:33.3259783 PM	powershell.exe	2224	Process Create	C:\Windows\system32\rundll32.exe	SUCCESS	PID: 3648, Command line: "C:\Windows\system32\rundll32.exe" C:\Windows\system32\shell32.dll,OpenAs_RunDLL C:\MinGW\msys\1.0\bin\cmd
2:14:33.5245525 PM	powershell.exe	2224	Process Create	C:\Windows\system32\rundll32.exe	SUCCESS	PID: 2844, Command line: "C:\Windows\system32\rundll32.exe" C:\Windows\system32\shell32.dll,OpenAs_RunDLL C:\MinGW\msys\1.0\bin\cmd
```

**Possible Fix:**
Do one of the following:

- use "where cmd.exe" command instead of "where cmd"
- use %COMSPEC% env variable instead of trying to locate cmd
- use explicit path to %SystemRoot%\system32\cmd.exe

**Comment by FeodorFitsner** [comment]
Fair enough, sounds like a bug. Right now `cmd` is called with `cmd /c ...` from PowerShell. Should be full path instead.
